fix: "Copy Image", "Open Image", "Save Image", "Open in Mask Editor" missing from context menu on SaveImage vue node

christian-byrne opened the pull request:
## Summary


| Before                                                                                                                                        | After                                                                                                                                         |
| --------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
| <img width="1559" height="1352" alt="Screenshot from 2025-12-12 05-22-16" src="https://github.com/user-attachments/assets/4baa60ac-8060-4188-9119-4a7f827b0f78" /> | <img width="1559" height="1352" alt="Screenshot from 2025-12-12 05-21-10" src="https://github.com/user-attachments/assets/aa9cae61-271d-4563-a0fa-f8bf619aaf8f" /> |


Fixes the "Copy Image" context menu option not working in Vue Nodes (Nodes 2.0). The issue was that `node.imgs` wasn't being populated in Vue nodes like it is in legacy nodes causing it to not include the image context menu options properly.

Added `syncNodeImgs()` to `ImagePreview.vue` that syncs the image element to `node.imgs` when images load, enabling backwards compatibility with legacy systems that rely on this property. Also simplified `setupNodeForMaskEditor()` since the sync now happens automatically. Added unit test for the sync behavior and browser test for mask editor button functionality.

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-7416-fix-Copy-Image-Open-Image-Save-Image-Open-in-Mask-Editor-missing-from-context-2c76d73d365081b8936fcaade7d2b15a) by [Unito](https://www.unito.io)

christian-byrne committed:
assign node images for backwards compat

christian-byrne committed:
simplify

DrJKL commented on browser_tests/tests/vueNodes/maskEditor.spec.ts:
```diff
@@ -0,0 +1,60 @@
+    await imageWrapper.hover()
+
+    const overlayMaskButton = comfyPage.page.locator(
+      '.image-preview [aria-label="Edit or mask image"]'
```
We could use [getByLabel](https://playwright.dev/docs/locators#locate-by-label) here

Commit:
[automated] Update test expectations

DrJKL commented on browser_tests/tests/nodeSearchBox.spec.ts-snapshots/added-node-no-connection-chromium-linux.png:
Seems unrelated, not sure what's up here.

christian-byrne committed:
fix: sync node.imgs in Vue nodes for Copy Image compatibility

- Add syncNodeImgs() to ImagePreview.vue to sync node.imgs on image load
- Simplify setupNodeForMaskEditor() to reuse sync logic
- Add aria-label to MaskEditorButton for accessibility
- Add browser test for mask editor button functionality
- Add unit test for node.imgs sync behavior

Fixes copy image and other legacy features that rely on node.imgs

christian-byrne committed:
Merge branch 'main' into image-output-backwards-compat

christian-byrne committed:
fix: use combo widget instead of drag-drop in mask editor test

christian-byrne committed:
fix: use existing workflow image instead of widget interaction

christian-byrne committed:
refactor: move node.imgs sync from component to store

Move syncNodeImgs logic from ImagePreview.vue to imagePreviewStore.ts
for proper separation of concerns. Store handles backwards compatibility
sync when setting node outputs.

christian-byrne committed:
fix: only sync node.imgs in Vue nodes mode

christian-byrne committed:
fix: add error handling and race condition guard to syncNodeImgs

christian-byrne committed:
fix: add missing afterEach import

christian-byrne committed:
fix: zoom out in mask editor test to keep toolbox in viewport

christian-byrne committed:
fix: use toBeHidden and getByLabel in mask editor test

christian-byrne committed:
fix: zoom out more in mask editor test
